ui/app/components/confirm-page-container/confirm-page-container-content/confirm-page-container-summary/confirm-page-container-summary.component.js

```diff
@@ -0,0 +1,56 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import classnames from 'classnames'
+import Identicon from '../../../identicon'
+
+const ConfirmPageContainerSummary = props => {
+  const { action, title, subtitle, hideSubtitle, className, identiconAddress, nonce } = props
+
+  return (
+    <div className={classnames('confirm-page-container-summary', className)}>
+      <div className="confirm-page-container-summary__action-row">
+        <div className="confirm-page-container-summary__action">
+          { action }
+        </div>
+        {
+          nonce && (
+            <div className="confirm-page-container-summary__nonce">
+              { `#${nonce}` }
+            </div>
+          )
+        }
+      </div>
+      <div className="confirm-page-container-summary__title">
+        {
+          identiconAddress && (
+            <Identicon
+              className="confirm-page-container-summary__identicon"
+              diameter={36}
+              address={identiconAddress}
+            />
+          )
+        }
+        <div className="confirm-page-container-summary__title-text">
+          { title }
+        </div>
+      </div>
+      {
+        hideSubtitle || <div className="confirm-page-container-summary__subtitle">
```

Comment:
It might be slight improvement on code readability if this was `showSubtitle && `. I say this because the other conditionally rendered elements are rendered if a value is truthy with `&&`. Reading the file and then encountering `||` is a little confusing. I say this lightly though, if switching to `showSubtitle &&` makes things less readable or intuitive elsewhere, then my suggestion here should be ignored.

Reply:
I did it this way because the default of the summary component is to show the subtitle, and hiding it is the exception. If I used `showSubtitle` as the boolean prop then I think it's a less convenient API since I'd have to add that in more places than not.